mm2src/coins/hd_wallet_storage/sqlite_storage.rs

```diff
@@ -0,0 +1,309 @@
+    (coin, mm2_rmd160, hd_wallet_rmd160, account_id, account_xpub, external_addresses_number, internal_addresses_number)
+    VALUES (?1, ?2, ?3, ?4, ?5, ?6, ?7);";
+
+const DELETE_ACCOUNTS_BY_WALLET_ID: &str =
+    "DELETE FROM hd_account WHERE coin=?1 AND mm2_rmd160=?2 AND hd_wallet_rmd160=?3;";
+
+const SELECT_ACCOUNT: &str = "SELECT account_id, account_xpub, external_addresses_number, internal_addresses_number
+    FROM hd_account
+    WHERE coin=?1 AND mm2_rmd160=?2 AND hd_wallet_rmd160=?3 AND account_id=?4;";
+
+const SELECT_ACCOUNTS_BY_WALLET_ID: &str =
+    "SELECT account_id, account_xpub, external_addresses_number, internal_addresses_number
+    FROM hd_account
+    WHERE coin=?1 AND mm2_rmd160=?2 AND hd_wallet_rmd160=?3;";
+
+/// The max number of SQL query params.
+const PARAMS_CAPACITY: usize = 7;
+
+impl From<SqlError> for HDWalletStorageError {
+    fn from(e: SqlError) -> Self {
+        let error = e.to_string();
+        match e {
+            SqlError::FromSqlConversionFailure(_, _, _)
+            | SqlError::IntegralValueOutOfRange(_, _)
+            | SqlError::InvalidColumnIndex(_)
+            | SqlError::InvalidColumnType(_, _, _) => HDWalletStorageError::ErrorDeserializing(error),
+            SqlError::Utf8Error(_) | SqlError::NulError(_) | SqlError::ToSqlConversionFailure(_) => {
+                HDWalletStorageError::ErrorSerializing(error)
+            },
+            _ => HDWalletStorageError::Internal(error),
+        }
+    }
+}
+
+impl TryFrom<&Row<'_>> for HDAccountStorageItem {
+    type Error = SqlError;
+
+    fn try_from(row: &Row<'_>) -> Result<Self, Self::Error> {
+        Ok(HDAccountStorageItem {
+            account_id: row.get(0)?,
+            account_xpub: row.get(1)?,
+            external_addresses_number: row.get(2)?,
+            internal_addresses_number: row.get(3)?,
+        })
+    }
+}
+
+impl HDAccountStorageItem {
+    fn to_sql_params_with_wallet_id(&self, wallet_id: HDWalletId) -> Vec<String> {
+        let mut params = Vec::with_capacity(PARAMS_CAPACITY);
+        wallet_id.fill_sql_params(&mut params);
+        self.fill_sql_params(&mut params);
+        params
+    }
+
+    fn fill_sql_params(&self, params: &mut Vec<String>) {
+        params.push(self.account_id.to_string());
+        params.push(self.account_xpub.clone());
+        params.push(self.external_addresses_number.to_string());
+        params.push(self.internal_addresses_number.to_string());
+    }
+}
+
+impl HDWalletId {
+    fn to_sql_params(&self) -> Vec<String> {
+        let mut params = Vec::with_capacity(PARAMS_CAPACITY);
+        self.fill_sql_params(&mut params);
+        params
+    }
+
+    fn fill_sql_params(&self, params: &mut Vec<String>) {
+        params.push(self.coin.clone());
+        params.push(self.mm2_rmd160.clone());
+        params.push(self.hd_wallet_rmd160.clone());
+    }
+}
+
+#[derive(Clone)]
+pub struct HDWalletSqliteStorage {
+    conn: SqliteConnWeak,
```

Comment:
Can we safely use `Arc` instead of `Weak` here?

Reply:
Yes, we can. But I think it's not necessary here to keep `Arc`, so I decided to keep a weak pointer instead